### ISSUE TITLE
`rust_analyzer` doesn't report some errors 

### ISSUE THREAD

**Issue by ribelo** [neovim/neovim#24427]
### Problem

tl:dr
`rust_analyzer` doesn't report some errors if we have the workspace as a dependency.

I came across this in my repository, but the simplest way to check is for example ripgrep. If we have a workspace defined in Cargo.toml as a dependency, then suddenly some errors are not reported.

### Steps to reproduce using "nvim -u minimal_init.lua"

1. `git clone https://github.com/LazyVim/starter ~/.config/nvim`
2. `vim ~/.config/nvim/lua/config/lazy.lua`
3. add plugin `{ import = "lazyvim.plugins.extras.lang.rust" },`
1. `git clone https://github.com/BurntSushi/ripgrep`
2. `vim ./crates/core/main.rs`
3. `53jj`
4. `i dbg!(foo + bar)`
5. `:w`

### Expected behavior

`foo` and `bar` are undefined and should be reported as an error.

### Neovim version (nvim -v)

NVIM v0.9.1

### Language server name/version

rust-analyzer 1.71.0

### Operating system/version

NixOS
